src/pages/projects/forms/NetworkSelector.tsx

```diff
@@ -0,0 +1,67 @@
+import { Select } from "@canonical/react-components";
+import type { SelectProps } from "@canonical/react-components";
+import type { FC } from "react";
+import { useNetworks } from "context/useNetworks";
+
+interface Props {
+  value: string;
+  project: string;
+  setValue: (value: string) => void;
+  selectProps?: SelectProps;
+  onBlur?: (e: React.FocusEvent) => void;
+  hasNoneOption?: boolean;
+}
+
+const NetworkSelector: FC<Props> = ({
+  value,
+  project,
+  setValue,
+  selectProps,
+  onBlur,
+  hasNoneOption = false,
+}) => {
+  const { data: networks = [] } = useNetworks(project);
+
+  const managedNetworks = networks.filter((network) => network.managed);
+
+  const getNetworkOptions = () => {
+    const options = managedNetworks.map((network) => {
+      return {
+        label: network.name,
+        value: network.name,
+        disabled: false,
+      };
+    });
+
+    options.unshift({
+      label: options.length === 0 ? "No networks available" : "Select option",
+      value: "",
+      disabled: true,
+    });
+
+    if (hasNoneOption) {
+      options.push({ label: "None", value: "none", disabled: false });
+    }
+
+    return options;
+  };
+
+  const handleNetworkChange = (value: string) => {
+    setValue(value);
+  };
+
+  return (
+    <Select
+      label="Network"
+      onChange={(e) => {
+        handleNetworkChange(e.target.value);
```

Comment:
```suggestion
        setValue(e.target.value);
```
We can remove the extra callback.